src/caughtup/__tests__/caughtUpReducer-test.js

```diff
@@ -65,6 +66,24 @@ describe('caughtUpReducer', () => {
 
       expect(newState).toEqual(expectedState);
     });
+
+    test('if fetched messages are from a search narrow, discard them', () => {
+      const initialState = deepFreeze({
+        [HOME_NARROW_STR]: {
+          older: false,
+          newer: false,
+        },
+      });
+
+      const action = deepFreeze({
+        ...eg.action.message_fetch_complete,
+        narrow: [{ operator: 'search', operand: 'some query' }],
+      });
+
+      const newState = caughtUpReducer(initialState, action);
+
+      expect(newState).toEqual(initialState);
+    });
```

Comment:
Reading a test, a key question to ask is: does this really test what it's meant to test? This is especially important for tests, like this one, that test that nothing happens -- it'd be easy to accidentally write it so that some other aspect of the test case prevents the thing from happening, and so the test would pass even if the logic it's meant to test was completely missing.

It's hard to tell from reading just this test whether that's the case, because it's not obvious whether the example value for this action type would be one that makes `caughtUp` become true or not.

In general, one good way to make that clear is: make the test case's input data look exactly like another test case that tests something *does* happen, except for just one difference, namely whatever it is that you want to test prevents the thing from happening.

Here, that'd mean taking one of the test cases with "true" in the expected new state, copying it, and just changing the narrow to a search narrow.

Reply:
> In general, one good way to make that clear is: make the test case's input data look exactly like another test case that tests something _does_ happen, except for just one difference, namely whatever it is that you want to test prevents the thing from happening.

Thanks -this is helpful!

Reply:
I've changed the tests for `caughtUpReducer`, would you like me to also change the tests for other reducers I've modified? That would also make the commit `example data: Add sample action 'message_fetch_complete'` un-necessary.

Reply:
@gnprice [said](https://github.com/zulip/zulip-mobile/pull/3868#discussion_r387931258):
> Here, that'd mean taking one of the test cases with "true" in the expected new state, copying it, and just changing the narrow to a search narrow.

In the status quo, this does sort of undo the progress toward getting more tests using exampleData. I wonder about, in a separate commit, getting all the existing tests in the file to use the action from `exampleData`, and then applying the change I've just quoted.

Reply:
FYI, there will be merge conflicts with anything that uses `exampleData` because of https://github.com/zulip/zulip-mobile/commit/50a07b8b9c9105f18958568a49b709e1cdce3328.

Reply:
> In the status quo, this does sort of undo the progress toward getting more tests using exampleData. I wonder about, in a separate commit, getting all the existing tests in the file to use the action from `exampleData`, and then applying the change I've just quoted. @gnprice, thoughts?

Yeah, I agree -- that combination would be ideal.

Doesn't need to mean getting the whole test files to be type-checked -- just using the new `eg.action.message_fetch_complete` in the places where it fits.

Reply:
Done.